app/services/signbank/sign_search_service.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+module Signbank
+  class SignSearchService
+    SEARCH_SQL = <<-SQL.squish
+    WITH sign_search(id, rank_precedence, rank_order, row_num) AS
```

Comment:
This is the same search implementation as NZSL Share, with additional filter searches for handshape, location, usage and topic tags